src/components/pages/IDPManagement/IDPList.tsx

```diff
@@ -94,9 +95,22 @@ export const IDPList = ({ isManagementOSP }: { isManagementOSP?: boolean }) => {
     .sort((a: IdentityProvider, b: IdentityProvider) =>
       (a?.displayName ?? '').localeCompare(b.displayName ?? '')
     )
+  const [searchExpr, setSearchExpr] = useState<string>('')
   const [removeIDP] = useRemoveIDPMutation()
   const [enableIDP] = useEnableIDPMutation()
 
+  const managedIdpsData = () => {
```

Comment:
@oyo  Isn't it adding more complexity by running loop multiple times?

Reply:
I prefer a const variable instead or using a react hook either useMemo() or useCallback() based on the type